docs/builder_image.md

```diff
@@ -11,26 +11,25 @@ of the two basic requirements for the best possible sti performance, these are:
 # Required image contents
 
 The build process consists of three fundamental elements which are combined into
-final docker image, the three are: sources, scripts and builder image. During the
+final docker image.  These three are: sources, scripts and builder image. During the
 build process sti must place sources and scripts inside that builder image. To do
 so sti creates a tar file containing the two and then streams that file into the
-builder image. Before executing `assemble` script, sti untars that file and places
-its contents into the destination specified with `--destination` flag or `io.s2i.destination`
+builder image. Before executing the `assemble` script, sti untars that file and places
+its contents into the destination specified with either the `--destination` flag or `io.s2i.destination`
 label from the builder image (default destination is `/tmp`). For this
-to happen your image must supply tar archiving utility (command `tar` available in `$PATH`)
-and command line interpreter (command `/bin/sh`). Doing so will allow your image to
+to happen your image must supply the tar archiving utility (command `tar` available in `$PATH`)
+and a command line interpreter (command `/bin/sh`). Doing so will allow your image to
 use the fastest possible build path, because in all other cases when either
-`tar` or `/bin/sh` is not available, sti build will be forced to perform an additional
+`tar` or `/bin/sh` are not available, sti build will be forced to perform an additional
 docker build to put both sources and scripts inside the image and only then run the
-usual sti build procedure (sti will do this automatically if tar and /bin/sh are not found).
+usual sti build procedure (sti will do this automatically if tar or /bin/sh are not found).
 See the following diagram for how build workflow looks like:
 
 ![sti workflow](./sti-flow.png "sti workflow")
 
 \* Run build's responsibility is to untar the sources, scripts and artifacts (if such
-exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh`
-error) it's responsible only for invoking `assemble` script, since both scripts and
-sources are already there.
+exist) and invoke the `assemble` script. If this is the second run (due to catching any `tar`/`/bin/sh` related
+errors) it's responsibility is only to invoke the `assemble` script, since both scripts and sources are already there.
```

Comment:
Whoops, it definitely should be.  I'll fix that in a PR today with some other updates I was planning on making to the CLI page.